### ISSUE TITLE
no commit hash url text

### ISSUE THREAD

**Issue by Slluxx** [metcalfc/changelog-generator#53]
Releases have this text for me:
```
* [](http://github.com/../../../) - Update VERSION
* [](http://github.com/../../../) - Update buildRelease.yml
* [](http://github.com/../../../) - Update buildRelease.yml
```

The URL works and is complete but the text is missing, thus its displaying a line as:

* `-` Update VERSION

I am building a release like this (in case this has to do something with it):

```
    - name: Create Release
      run: |
        set -x
        assets=()
        for asset in ./*.zip; do
          assets+=("-a" "$asset")
        done
        hub release create "${assets[@]}" -d -m "Release v${{ steps.set-version.outputs.version }}
        
        ${{ steps.changelog.outputs.changelog }}" "v${{ steps.set-version.outputs.version }}"
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by metcalfc** [comment]
Thats odd. So the bit that does the real work here is a one liner in bash:

```
log=$(git log "${base_ref}...${head_ref}" \
  --pretty=format:"* [\`%h\`](http://github.com/${repo_url}/commit/%H) - %s" \
  --reverse)
```

My guess is that the way you're doing the notes is getting wrecked in the shell via expansions. There are backticks in the short hash link so those could be getting evaluated. I'm not sure if there is a better way to escape those in a shell.

Have you considered using a release specific action instead of hub in shell script? https://github.com/marketplace/actions/create-release seems reasonably aligned:

```
    - uses: ncipollo/release-action@v1
      with:
        artifacts: "./*zip"
        name: "Release v${{ steps.set-version.outputs.version }}"
        body: "${{ steps.changelog.outputs.changelog }}"
        tag: "v${{ steps.set-version.outputs.version }}"
        token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by Slluxx** [comment]
Allrighty, using release-action@v1 the body looks just like expected. I had a reason for using hub but i think that obsolete so this solution is 100% better.


I do have 2 unrelated feature requests though:
Is it possible to display not just the commit title but also the description? like this:

- (0101010)[Merge Pull request #52 ] - Added callback to system method 
    And here some text. more explanation without cramming important sentences into the title.
    Preferably multi line, for some beautiful formatted text. maybe the bodys commit-message can have markdown styling so it shows here perfectly?
- (1010101)[Another commit] - very important short description
- (1111101)[Another commit 2] - very important short description 2
    again with body text
    
And my other, more advanced request would be to use keywords to group commit messages.
See this example: https://github.com/olivierlacan/keep-a-changelog/blob/master/CHANGELOG.md

Basically if a commit message starts with "Added, Changed, Removed, Fixed" it will group them together. Forget about the date and versioning in the example. Just a header, starting with "###", the keyword as title and a list of the grouped messages. If that would then also include the commit description just like in my first feature request, this changelog generator would be absolutely perfect!

Thanks for your work and time and help no matter if you like my idea or not!

**Comment by Slluxx** [comment]
@metcalfc hey you havent replied in a while. i know my text would belong in a new issue but may you still take a look at my suggestion?

**Comment by metcalfc** [comment]
Hey Calvin, sorry for the late reply. I'm on an extended vacation and doing
my best to stay off work related sites.

What I set out for this action was to be the most simple, collect the
commits since the last release, configure next to nothing, and don't
require commits to be a format action in its category.  I think those are
good features. And there are a few similar actions that implement them
like: Release Drafter · Actions · GitHub Marketplace
<https://github.com/marketplace/actions/release-drafter>

GitHub marketplace doesn't do a great job surfacing all the options and
making it clear what the feature differences are. Maybe thats a good thing
to add to the README. "This is the super simple changelog generator"

I appreciate the feedback.

**Comment by Slluxx** [comment]
Sorry to disturb your vacation! Your action is great as is and i completely get the point of a simple one like this. Thanks for pointing me in another direction and i hope you have a few more great days ahead! 

Cheers!
